libr/reg/value.c

```diff
@@ -2,6 +2,14 @@
 
 #include <r_reg.h>
 #include <r_util.h>
+#include <r_types.h>
+
+#define SAME_ENDIAN ( \
+	   ((strcmp(R_SYS_ENDIAN, "little") == 0) && reg->big_endian == 0) \
```

Comment:
compare numbers instead of strings, this check can be done by cpp preprocessor, no need to do this at runtime